### ISSUE TITLE
ArgumentNullException when running RoslynDeployment

### ISSUE THREAD

**Issue by YairHalberstadt** [dotnet/roslyn#35889]
**Version Used**: 

https://github.com/dotnet/roslyn/pull/35539
VS 16.1

**Steps to Reproduce**:

When I run RoslynDeployment with the debugger attached (I haven't tried without), and once VS is loaded I open an existing solution, I get the following exception:

```
System.ArgumentNullException
  HResult=0x80004003
  Message=Value cannot be null.
Parameter name: key
  Source=mscorlib
  StackTrace:
Microsoft.VisualStudio.LanguageServices.dll!Microsoft.VisualStudio.LanguageServices.Implementation.Venus.ContainedDocument.TryGetContainedDocument(Microsoft.CodeAnalysis.DocumentId id) Line 66
	at E:\Git\roslyn\src\VisualStudio\Core\Def\Implementation\Venus\ContainedDocument.cs(66)
Microsoft.VisualStudio.LanguageServices.dll!Microsoft.VisualStudio.LanguageServices.Implementation.SuggestionService.VisualStudioSupportsFeatureService.SupportsCodeFixesWorker(Microsoft.CodeAnalysis.DocumentId id) Line 87
	at E:\Git\roslyn\src\VisualStudio\Core\Def\Implementation\VisualStudioSupportsFeatureService.cs(87)
Microsoft.VisualStudio.LanguageServices.dll!Microsoft.VisualStudio.LanguageServices.Implementation.SuggestionService.VisualStudioSupportsFeatureService.VisualStudioTextBufferSupportsFeatureService.SupportsCodeFixes(Microsoft.VisualStudio.Text.ITextBuffer textBuffer) Line 24
	at E:\Git\roslyn\src\VisualStudio\Core\Def\Implementation\VisualStudioSupportsFeatureService.cs(24)
Microsoft.CodeAnalysis.EditorFeatures.dll!Microsoft.CodeAnalysis.Editor.Shared.Extensions.ITextBufferExtensions.SupportsCodeFixes(Microsoft.VisualStudio.Text.ITextBuffer buffer) Line 66
	at E:\Git\roslyn\src\EditorFeatures\Core\Shared\Extensions\ITextBufferExtensions.cs(66)
Microsoft.CodeAnalysis.EditorFeatures.Wpf.dll!Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.SuggestedActionsSourceProvider.SuggestedActionsSource.GetFixLevelAsync(Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.SuggestedActionsSourceProvider provider, Microsoft.CodeAnalysis.Document document, Microsoft.VisualStudio.Text.SnapshotSpan range, System.Threading.CancellationToken cancellationToken) Line 806
	at E:\Git\roslyn\src\EditorFeatures\Core.Wpf\Suggestions\SuggestedActionsSource.cs(806)
Microsoft.CodeAnalysis.EditorFeatures.Wpf.dll!Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.SuggestedActionsSourceProvider.SuggestedActionsSource.GetSuggestedActionCategoriesAsync.AnonymousMethod__0() Line 1008
	at E:\Git\roslyn\src\EditorFeatures\Core.Wpf\Suggestions\SuggestedActionsSource.cs(1008)
```

It looks like we pass the result of `VisualStudioSupportsFeatureService.GetContainedDocumentId()`, which can return null, to `SupportsCodeFixesWorker`, which cannot accept null.

A bunch of errors also occur after this. I don't know if that is correlation or causation though. However roslyn fails to spot any semantic errors in the document after this.

**Comment by jinujoseph** [comment]
cc @sharwell

**Comment by dibarbet** [comment]
Thanks for the report!
So this should be fixed in https://github.com/dotnet/roslyn/pull/35831 because we saw this issue elsewhere (in liveshare).  However I'm not able to reproduce your specific case and confirm the fix there.

@YairHalberstadt would you be able to either 
1)  Check if you can reproduce it yourself with the commit I mentioned above pulled in or
2)  Help me figure out how to figure out exact reproduction steps?

The steps I'm following - 
1.  Find commit prior to when this was merged in https://github.com/dotnet/roslyn/pull/35831
2.  Open Roslyn.sln in VS 16.1.0 on that commit (I picked #a7c98c)
3.  F5 RoslynDeployment
4.  Open a solution in the VS being debugged.

After 4 I should expect to see that exception, but I'm not seeing anything.
A few things that might help - the commit of Roslyn you're using, the type of solution you're opening (e.g. is it an ASP.net application, does it have .cshtml files or maybe a blazor project)?, and any other details about the specific steps you're following.

Thanks :)

**Comment by YairHalberstadt** [comment]
Ok I've merged the latest master into #35539, and I'm no longer getting that particular exception. 

I continued getting other errors when opening the solution I opened previously. However when I restarted and opened a different solution, then restarted and opened my original solution, it seemed to work fine.

I'll close this as it sounds like it's going to be impossible to reproduce, and it's working fine now. My main reason for opening the issue was the inconsistency between `GetContainedDocumentId()` returning null and `SupportsCodeFixesWorker` not accepting null, but that is fixed now.
